src/include/osl_pvt.h

```diff
@@ -205,7 +205,15 @@ class TypeSpec {
 
     /// Returns the length of the array, or 0 if not an array.
     ///
-    int arraylength () const { return m_simple.arraylen; }
+    int arraylength () const { return abs(m_simple.arraylen); }
```

Comment:
Per my comments on the OSL side -- would it make sense to instead keep the OIIO semantics the same, but add a field in here to indicate the default varlen size?